src/agent/src/mount.rs

```diff
@@ -405,14 +405,18 @@ async fn bind_watcher_storage_handler(
     logger: &Logger,
     storage: &Storage,
     sandbox: Arc<Mutex<Sandbox>>,
+    cid: Option<String>,
 ) -> Result<()> {
     let mut locked = sandbox.lock().await;
-    let container_id = locked.id.clone();
 
-    locked
-        .bind_watcher
-        .add_container(container_id, iter::once(storage.clone()), logger)
-        .await
+    if let Some(cid) = cid {
+        locked
+            .bind_watcher
+            .add_container(cid, iter::once(storage.clone()), logger)
+            .await
+    } else {
+        Ok(())
```

Comment:
I am not fully familiar with the watcher logic please correct me if I am wrong.

When `none` is provided this becomes a  `no-op`.  Is this correct? Should not fail. If it is none is because the request is from `create_sandbox()` where containers does not exit. So probably the sandbox request may be not  well formed. In case want to fail silently, probably logging that this is ignored may be a good option.

Reply:
Hey @jcvenegas! 

 I expect that this handler really should only be invoked from the create_container() call, and in that scenario we should (always) have a cid. @liubin WDYT?
 
 Since, AFAIU, this is unexpected, I agree with at least logging here?

Reply:
@egernst @jcvenegas 
Yes, this function is called from create_container and create_sandbox, only the create_container should pass the cid.